lib/auth.guard.ts

```diff
@@ -22,14 +22,15 @@ export type IAuthGuard = CanActivate & {
   ): Promise<void>;
   handleRequest<TUser = any>(err, user, info, context, status?): TUser;
   getAuthenticateOptions(context): IAuthModuleOptions | undefined;
+  getRequest<T = any>(context: ExecutionContext): T;
 };
 export const AuthGuard: (
   type?: string | string[]
 ) => Type<IAuthGuard> = memoize(createAuthGuard);
 
 const NO_STRATEGY_ERROR = `In order to use "defaultStrategy", please, ensure to import PassportModule in each place where AuthGuard() is being used. Otherwise, passport won't work correctly.`;
 
-function createAuthGuard(type?: string | string[]): Type<CanActivate> {
+function createAuthGuard(type?: string | string[]): Type<IAuthGuard> {
```

Comment:
Is this really necessary? I think adding the type directly in the IAuthGuard type definition is enough.